model/src/application/interaction/application_command/option.rs

```diff
@@ -459,4 +601,35 @@ mod tests {
             ],
         );
     }
+
+    #[test]
+    fn autocomplete() {
+        let value = CommandDataOption {
+            name: "opt".to_string(),
+            value: CommandOptionValue::Focused(
+                "not a number".to_owned(),
```

Comment:
This is something that wouldn't pass Discord's client side verification right? I.e. only a malicious user should be able to send it?

Wouldn't erroring on this be fine? It would improve the API

Reply:
While being focused users can enter anything (even meaningless strings) and Discord sends it (Discord does not validate the focused field).